### PR TITLE
Fetch servers from HMW master

### PR DESCRIPTION
**Pull request by tobibodamer**
Use HTTP getInfo for HMW servers only
Dynamic server info request implementation based on server type

### CODE REVIEW

File: H2MLauncher.Core/Networking/GameServer/HMW/HttpEventListener.cs [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+            else if (eventSource.Name == "System.Threading.Tasks.TplEventSource")
+            {
+                // Attach ActivityId to the events.
+                EnableEvents(eventSource, EventLevel.LogAlways, TasksFlowActivityIds);
+            }
+        }
+
+        protected override void OnEventWritten(EventWrittenEventArgs eventData)
+        {
+            var timings = _timings.Value;
```
Comment (Bowhza) [inline H2MLauncher.Core/Networking/GameServer/HMW/HttpEventListener.cs:36]: Avoid using var

File: H2MLauncher.Core/Networking/GameServer/HMW/HttpEventListener.cs [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+                    break;
+                case "System.Net.Http.ResponseContentStop":
+                    timings.ResponseContentStop = eventData.TimeStamp;
+                    break;
+            }
+        }
+
+        public HttpRequestTimings GetTimings()
+        {
+            var raw = _timings.Value!;
```
Comment (Bowhza) [inline H2MLauncher.Core/Networking/GameServer/HMW/HttpEventListener.cs:93]: here too

File: H2MLauncher.Core/Networking/GameServer/HMW/HttpGameServerCommunicationService.cs [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+            Channel<(TServer server, GameServerInfo? info)> channel = Channel.CreateUnbounded<(TServer, GameServerInfo?)>();
+            ConcurrentBag<Task> continuations = [];
+
+            Task requestTask = Parallel.ForEachAsync(servers, new ParallelOptions() { CancellationToken = cancellationToken }, async (server, ct) =>
+            {
+                CancellationTokenSource timeoutCancellation = new(requestTimeoutInMs);
+                CancellationTokenSource linkedCancellation = CancellationTokenSource.CreateLinkedTokenSource(timeoutCancellation.Token, ct);
+                try
+                {
+                    var task = await GetInfoCoreAsync(server, linkedCancellation.Token);
```
Comment (Bowhza) [inline H2MLauncher.Core/Networking/GameServer/HMW/HttpGameServerCommunicationService.cs:55]: var here

File: H2MLauncher.Core/Networking/CachedIpv6EndpointResolver.cs [hunk 4]
```diff
@@ -76,5 +79,35 @@ private record struct IpEndpointCacheKey(string IpOrHostName, int Port) { }
...
+                new ParallelOptions()
+                {
+                    CancellationToken = cancellationToken,
+                    MaxDegreeOfParallelism = 20
+                },
+                async (server, token) =>
+                {
+                    // create an endpoint to send to and receive from
+                    IPEndPoint? endpoint = await GetEndpointAsync(server, token);
+                    if (endpoint != null)
```
Comment (metalglove) [inline H2MLauncher.Core/Networking/CachedIpv6EndpointResolver.cs:100]: Reduce indentation return early

File: H2MLauncher.Core/Networking/GameServer/HMW/HttpGameServerCommunicationService.cs [hunk 5]
```diff
@@ -0,0 +1,247 @@
...
+                    Port = server.Port,
+                    Path = "getInfo"
+                };
+
+                Uri url = uriBuilder.Uri;
+
+                HttpResponseMessage response;
+                HttpEventListener.HttpRequestTimings timings;
+
+                //_httpClient.DefaultRequestHeaders.CacheControl = new CacheControlHeaderValue
```
Comment (metalglove) [inline H2MLauncher.Core/Networking/GameServer/HMW/HttpGameServerCommunicationService.cs:218]: Comment?

File: H2MLauncher.Core/Networking/GameServer/ICanGetGameServerInfo.cs [hunk 6]
```diff
@@ -0,0 +1,15 @@
+﻿using H2MLauncher.Core.Models;
+using H2MLauncher.Core.Networking.GameServer;
+
+namespace H2MLauncher.Core.Services
+{
+    public interface ICanGetGameServerInfo<TServer> where TServer : IServerConnectionDetails
```
Comment (metalglove) [inline H2MLauncher.Core/Networking/GameServer/ICanGetGameServerInfo.cs:6]: Find suitable name

**Review by metalglove — CHANGES_REQUESTED** [review]

**Pull request merged**